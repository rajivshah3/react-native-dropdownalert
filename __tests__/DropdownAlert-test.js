import reactNative from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DropdownAlert from '../DropdownAlert';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
const imageSrc = 'https://facebook.github.io/react-native/docs/assets/favicon.png';

test('renders info alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({ isOpen: true, type: 'info', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders warn alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({ isOpen: true, type: 'warn', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders error alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({ isOpen: true, type: 'error', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders success alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({ isOpen: true, type: 'success', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders custom alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({ isOpen: true, type: 'custom', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders custom alert with cancel correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} showCancel={true} />);
  wrapper.setState({ isOpen: true, type: 'custom', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders alert with zIndex correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} zIndex={99} />);
  wrapper.setState({ isOpen: true, type: 'info', title: 'Hello', message: 'World!' });
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('expect state variables to change with replace disabled', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} replaceEnabled={false} closeInterval={4000} />);
  wrapper.instance().isOpen = false;
  wrapper.instance()._closeTimeoutId = setTimeout(function() {});
  wrapper.update();
  wrapper.instance().alertWithType('info', 'hello', 'world');
  expect(wrapper.instance().state.type).toBe('info');
  expect(wrapper.instance().state.title).toBe('hello');
  expect(wrapper.instance().state.message).toBe('world');
  expect(wrapper.instance().state.isOpen).toBeTruthy();
  expect(wrapper.instance().state.topValue).toBe(0);
  expect(wrapper.instance()._closeTimeoutId).toBeDefined();
  expect(wrapper.instance().state.animationValue).toEqual(new reactNative.Animated.Value(0));
});
test('expect invalid type with state isOpen to be false', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.instance().alertWithType('random', 'hello', 'world');
  expect(wrapper.instance().state.isOpen).toBeFalsy();
});
test('expect non string title and message to converted to string', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.instance().alertWithType('error', {title: 'hello'}, {message: 'world'});
  console.log(wrapper.instance().state.title);
  expect(typeof wrapper.instance().state.title === 'string').toBeTruthy();
  expect(typeof wrapper.instance().state.message === 'string').toBeTruthy();
});
test('expect close without action to be animation value 0', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({isOpen: true});
  wrapper.instance().close();
  expect(wrapper.instance().state.isOpen).toBeTruthy();
});
test('expect close with action to be animation value 0', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.instance().close('cancel');
  expect(wrapper.instance().state.animationValue).toEqual(new reactNative.Animated.Value(0));
});
test('expect close directly to set isOpen false', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={imageSrc} />);
  wrapper.setState({isOpen: true});
  wrapper.instance()._closeTimeoutId = setTimeout(function() {});
  wrapper.update();
  wrapper.instance().closeDirectly();
  expect(wrapper.instance().state.isOpen).toBeFalsy();
  expect(wrapper.instance()._closeTimeoutId).toBeDefined();
});