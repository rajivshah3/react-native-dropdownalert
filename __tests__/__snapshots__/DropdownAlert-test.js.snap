// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders alert with zIndex correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
      "zIndex": 99,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#2B73B6",
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders custom alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "flexDirection": "row",
            "padding": 16,
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source="https://facebook.github.io/react/img/logo_og.png"
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders custom alert with cancel correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "flexDirection": "row",
            "padding": 16,
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source="https://facebook.github.io/react/img/logo_og.png"
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "width": 36,
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
      </TouchableOpacity>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders error alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#cc3232",
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders info alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#2B73B6",
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders success alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#32A54A",
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;

exports[`renders warn alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "elevation": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.95}
    disabled={false}
    focusedOpacity={0.7}
    onLayout={[Function]}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#cd853f",
          },
        ]
      }
    >
      <undefined
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <ImageView
          imageProps={Object {}}
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "padding": 8,
            }
          }
        >
          <Label
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "textAlign": "left",
              }
            }
            text="Hello"
            textProps={Object {}}
          />
          <Label
            numberOfLines={3}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
            text="World!"
            textProps={Object {}}
          />
        </View>
      </undefined>
    </View>
  </TouchableOpacity>
</AnimatedComponent>
`;
